helix-term/src/ui/editor.rs

```diff
@@ -122,8 +125,25 @@ impl EditorView {
         // TODO: range doesn't actually restrict source, just highlight range
         let highlights: Vec<_> = match doc.syntax() {
             Some(syntax) => {
+                let scopes = theme.scopes();
                 syntax
-                    .highlight_iter(text.slice(..), Some(range), None, |_| None)
+                    .highlight_iter(text.slice(..), Some(range), None, |language| {
+                        loader
+                                .language_config_for_scope(&format!("source.{}", language))
+                                .and_then(|language_config| {
+                                    let config = language_config.highlight_config(scopes)?;
+                                    let config_ref = config.as_ref();
+                                    // Safe because the referenced `HighlightConfiguration` is behind
```

Comment:
I prefer you use the safety comment I gave you :P using `SAFETY:` also makes it a lot easier to grep for in code and is the standard in rustc.

i.e. https://github.com/rust-lang/rust/blob/da7ada584a3e5c3467b8b9f344b4a0f2a81ce32a/library/core/src/num/nonzero.rs#L203